### PR TITLE
SONARJAVA-595 Fix parsing error with empty statements in imports

### PR DESCRIPTION
**Pull request by m-g-sonar**

### CODE REVIEW

**Comment by SonarTech** [comment]
SonarQube analysis reported no new issues.

**Comment by benzonico** [comment]
This solution, while fixing parsing issue, does not provide all the information to the syntax tree. For instance it won't allow to implement a rule to detect emtpy statement in imports.

File: java-squid/src/main/java/org/sonar/java/model/JavaTree.java [hunk 1]
```diff
@@ -156,11 +157,11 @@ public boolean isLeaf() {
   public static class CompilationUnitTreeImpl extends JavaTree implements CompilationUnitTree {
     @Nullable
     private final ExpressionTree packageName;
-    private final List<ImportTree> imports;
+    private final List<ImportClauseTree> imports;
     private final List<Tree> types;
     private final List<AnnotationTree> packageAnnotations;
 
-    public CompilationUnitTreeImpl(@Nullable ExpressionTree packageName, List<ImportTree> imports,
+    public CompilationUnitTreeImpl(@Nullable ExpressionTree packageName, List<ImportClauseTree> imports,
```
Comment (SonarTech) [inline java-squid/src/main/java/org/sonar/java/model/JavaTree.java]: MAJORorg/sonar/java/model/JavaTree$CompilationUnitTreeImpl must not use com/sonar/sslr/api/AstNode[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AAstNode_SSLR_class_should_not_be_used)

File: java-squid/src/main/java/org/sonar/java/ast/parser/TreeFactory.java [hunk 2]
```diff
@@ -187,6 +197,10 @@ public ExpressionTree newPackageDeclaration(Optional<List<AnnotationTreeImpl>> a
     return (ExpressionTree) partial;
   }
 
+  public ImportClauseTree newEmptyImport(AstNode semicolonTokenAstNode) {
+    return (ImportClauseTree) new EmptyStatementTreeImpl(semicolonTokenAstNode);
```
Comment (SonarTech) [inline java-squid/src/main/java/org/sonar/java/ast/parser/TreeFactory.java]: :new: MINORRemove this unnecessary cast to "ImportClauseTree".[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1905)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 new issue:
* 1 minor

Details:
* [Remove this unnecessary cast to "ImportClauseTree".](https://github.com/SonarSource/sonar-java/blob/ae0348785d75796d54e9fc3b7f3071320a9240a2/java-squid/src/main/java/org/sonar/java/ast/parser/TreeFactory.java#L201)[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1905)

**Comment by SonarTech** [comment]
SonarQube analysis reported no new issues.

**Comment by benzonico** [comment]
Main problem with this is breaking change in the tree that will break some custom rules.  I am not found of the other solution where it leads to make some field of ImportTree nullable. 
So LGTM.

**Pull request closed without merging**